client_app/src/components/shelter/RepeatableShifts.js

```diff
@@ -0,0 +1,195 @@
+// client_app/src/components/shelter/RepeatableShifts.js
+
+import React, { useState } from "react";
+import { useParams } from "react-router-dom";
+import "../../styles/shelter/RepeatableShifts.css";
+
+const RepeatableShifts = () => {
+  const { shelterId } = useParams(); // grab the shelterId from URL
+
+  const [shifts, setShifts] = useState([]);
+  const [newShift, setNewShift] = useState({
+    shiftName: "",
+    startTime: "",
+    endTime: "",
+    requiredVolunteers: 1,
+    maxVolunteers: 5,
+  });
+  const [successMessage, setSuccessMessage] = useState("");
+
+  // Convert HH:MM to milliseconds since midnight
+  const timeStringToMillis = (timeStr) => {
+    const [hours, minutes] = timeStr.split(":").map(Number);
+    return (hours * 60 + minutes) * 60 * 1000;
+  };
+
+  const handleInputChange = (e) => {
+    const { name, value } = e.target;
+    setNewShift({ ...newShift, [name]: value });
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (newShift.startTime && newShift.endTime) {
+      const newEntry = {
+        ...newShift,
+        shift_start: timeStringToMillis(newShift.startTime),
+        shift_end: timeStringToMillis(newShift.endTime),
+      };
+      setShifts([...shifts, newEntry]);
+      setNewShift({
+        shiftName: "",
+        startTime: "",
+        endTime: "",
+        requiredVolunteers: 1,
+        maxVolunteers: 5,
+      });
+      setSuccessMessage("Shift added successfully!");
+      setTimeout(() => setSuccessMessage(""), 3000); // Remove after 3 sec
```

Comment:
I think the success message here is a bit early, since the shifts are just added to the list to be submitted (they are not submitted to the backend yet).